lib/trino-parquet/src/main/java/io/trino/parquet/predicate/TupleDomainParquetPredicate.java

```diff
@@ -695,4 +746,41 @@ private Function<Integer, Object> getConverter(PrimitiveType primitiveType)
             super(columnPath, Integer.class);
         }
     }
+
+    /**
+     * Check if the predicateValue might be in the bloomfilter
+     *
+     * @param bloomFilter parquet bloomfilter.
+     * @param predicateValue effective discrete predicate value.
+     * @param sqlType Type that contains information about the type schema from connector's metadata
+     * @return true if the predicateValue might be in the bloomfilter, false if the predicateValue absolutely is not in the bloomfilter
+     */
+    @VisibleForTesting
+    public static boolean checkInBloomFilter(BloomFilter bloomFilter, Object predicateValue, Type sqlType)
+    {
+        // todo: support TIMESTAMP, and DECIMAL
+        if (sqlType == TINYINT || sqlType == SMALLINT || sqlType == INTEGER || sqlType == BIGINT || sqlType == DATE) {
+            return bloomFilter.findHash(bloomFilter.hash((int) asLong(predicateValue)));
+        }
+        else if (sqlType == DOUBLE) {
+            return bloomFilter.findHash(bloomFilter.hash(predicateValue));
+        }
+        else if (sqlType == REAL) {
+            return bloomFilter.findHash(bloomFilter.hash(intBitsToFloat(toIntExact(((Number) predicateValue).intValue()))));
```

Comment:
Thanks, changed to: 
`bloomFilter.hash(intBitsToFloat(toIntExact(((Number) predicateValue).longValue())))`

I referred implementation at RealValueWrite.java:

`float value = intBitsToFloat(toIntExact(REAL.getLong(block, i)));`

Reply:
I verified, and added test case for various extreme value, such as,  Float.POSITIVE_INFINITY, Float.NEGATIVE_INFINITY, Float.MIN_NORMAL, Float.MAX_VALUE was not able to reproduce precision lost error(when changed the casting from longValue() to intValue()). It is because float use 32 bits. 

But changed the implementation anyway.